app/models/concerns/tokenable.rb

```diff
@@ -0,0 +1,16 @@
+module Tokenable
+  extend ActiveSupport::Concern
+
+  included do
+    before_create :set_token
+  end
+
+  protected
+
+  def set_token
+    self.token = loop do
+      random_token = SecureRandom.hex(3)
```

Comment:
if you want unique random one you can use UUID - it's a standard for global unique ID - 

```ruby
>> SecureRandom.uuid
=> "30145f84-4c2d-42c9-9e36-c2291788e389"
```